File: encryption_example/Encryption.rb
require 'uri'

require "openssl"

def get_signature(param_str, secret)
  OpenSSL::HMAC.hexdigest('sha256', secret, param_str)
end

api_key = "B2Rou0PLPpGqcU0Vu2"
secret = "t7T0YlFnYXk0Fx3JswQsDrViLg1Gh3DUU5Mr"

params = {
  symbol: "BTCUSD",
  timestamp: Time.now.to_i * 1000,
  leverage: 100,
}.merge(api_key: api_key)

signature = get_signature(URI.encode_www_form(params.sort), secret)
puts signature

File: encryption_example/Encryption.js
const qs = require('qs')
const cryptojs = require('crypto-js')

var apiKey = "";
var secret = "";
var timestamp = Date.now();
var params = {
	"order_id":"876b0ac1-bafe-4110-b404-6a7c8211a6d9",
	"symbol":"BTCUSD",
	"timestamp":timestamp,
	"api_key" : apiKey,
};

console.log(getSignature(params, secret));

function getSignature(params, secret) {
  const payload = qs.stringify(params, { sort: (a, b) => a.localeCompare(b) })
  return cryptojs.HmacSHA256(payload, secret).toString(cryptojs.enc.Hex)
}

